app/components/Countdowntimer.jsx

```diff
@@ -0,0 +1,43 @@
+'use client';
+
+import PropTypes from 'prop-types';
+
+function CountdownTimer({ duration, onTimeUp }) {
+  const [timeLeft, setTimeLeft] = useState(duration);
+
+  useEffect(() => {
+    const timer = setInterval(() => {
+      setTimeLeft((prev) => {
+        if (prev <= 1) {
+          clearInterval(timer);
+          onTimeUp(navigate('/result'));
+          return 0;
+        }
+        return prev - 1;
+      });
+    }, 1000);
+
+    return () => clearInterval(timer);
+  }, [onTimeUp]);
+
+  const formatTime = (seconds) => {
+    const minutes = Math.floor(seconds / 60);
+    const secs = seconds % 60;
+    return `${minutes.toString().padStart(2, '0')}:${secs
+      .toString()
+      .padStart(2, '0')}`;
+  };
```

Comment:
You don't need to have `formatTime` this inside the component, please make it separate function